### PR TITLE
Feature#us#24/g rpc drug purchase

### PR DESCRIPTION
**Pull request by Branimirius**

### CODE REVIEW

File: src/HospitalAPI/HospitalAPI/HospitalAPI/Controllers/DrugPurchaseController.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+
+        [HttpPut("urgent")]
+        public IActionResult UrgentPurchase(DrugAmountDemandDto demand)
+        {
+            medicineService.AddDrugUrgent(demand.Name, demand.Amount);
+            return Ok(true);                         
+        }       
+        //COKA
```
Comment (lazar-stojcevic) [inline src/HospitalAPI/HospitalAPI/HospitalAPI/Controllers/DrugPurchaseController.cs:37]: COKACOLA

**Review by lazar-stojcevic — APPROVED** [review]
Very nice!

File: src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/GRPC/DrugDemandServiceGrpc.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        private MyDbContext dbContext;
+              
+        private MedicineService medicineService;
+        public HospitalService hospitalService;
+        private readonly ILogger<GreeterService> _logger;
+
+        private void SetupDbContext()
+        {
+            DbContextOptionsBuilder<MyDbContext> builder = new DbContextOptionsBuilder<MyDbContext>();
+            builder.UseNpgsql("server=localhost; port=5432; database=drugstore; User Id=postgres; password=firma4");
```
Comment (aleksandarbrkljac) [inline src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/GRPC/DrugDemandServiceGrpc.cs:25]: ??

File: src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Startup.cs [hunk 3]
```diff
@@ -53,20 +58,39 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
-            var port = Configuration["DBPort"];
-            var user = Configuration["DBUser"];
-            var password = Configuration["DBPassword"];
-            var database = Configuration["DB"];
+            var server = Configuration["DBServer"] ?? "localhost";
+            var port = Configuration["DBPort"] ?? "5432";
+            var user = Configuration["DBUser"] ?? "postgres";
+            var password = Configuration["DBPassword"] ?? "firma4";
+            var database = Configuration["DB"] ?? "drugstore";
             if (server == null) return ConfigurationExtensions.GetConnectionString(Configuration, "MyDbContextConnectionString");
```
Comment (aleksandarbrkljac) [inline src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Startup.cs:94]: If you are using conditional operator, this is not necessary.

File: src/Integration API/Controllers/DrugPurchaseController.cs [hunk 4]
```diff
@@ -97,5 +115,55 @@ private void SetApiKeyInHeader(DrugAmountDemandDto demand, RestRequest request)
...
+            }
+            return ApiKey;
+            
+        }
+
+        public bool drugDemandGrpc(DrugAmountDemandDto demand)
+        {
+            //var input = new HelloRequest { Name = "world" };
+            var input = new DrugRequest { Amount = demand.Amount, Name = demand.Name, PharmacyUrl = demand.PharmacyUrl, ApiKey = GetApiKey(demand) };
+            var channel = new Channel("127.0.0.1:4111", ChannelCredentials.Insecure);
```
Comment (aleksandarbrkljac) [inline src/Integration API/Controllers/DrugPurchaseController.cs:138]: ??

Comment (Branimirius) [reply src/Integration API/Controllers/DrugPurchaseController.cs:138]: Ovo smo zaboravili i sad sam ga sredio nadam se

File: src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/GRPC/DrugDemandServiceGrpc.cs (hunk 2 above)
Comment (Branimirius) [reply src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/GRPC/DrugDemandServiceGrpc.cs:25]: Moralo je ovako :((
Svi tako rade, bukvalno je nemoguce

**Pull request closed without merging**